### PR TITLE
[YUNIKORN-140] Create helm chart repository and publish it to helmhub

### PR DESCRIPTION
**Pull request by kingamarton**

### CODE REVIEW

File: helm-charts/index.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# TODO: update this file after releasing the chart
```
Comment (yangwwei) [inline helm-charts/index.yaml:1]: pls include apache license header

File: helm-charts/index.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+apiVersion: v1
+entries:
+  yunikorn:
+  - apiVersion: v1
+    appVersion: 0.1.0
```
Comment (yangwwei) [inline helm-charts/index.yaml:6]: what does the appVersion mean here?
how is it related to the release version?

File: helm-charts/yunikorn/.helmignore [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Patterns to ignore when building packages.
```
Comment (yangwwei) [inline helm-charts/yunikorn/.helmignore:1]: add apache license header

File: helm-charts/yunikorn/templates/cleanup.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+apiVersion: batch/v1
```
Comment (yangwwei) [inline helm-charts/yunikorn/templates/cleanup.yaml:1]: add apache license header

File: helm-charts/yunikorn/templates/configmap.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
```
Comment (yangwwei) [inline helm-charts/yunikorn/templates/configmap.yaml:1]: add apache license header

File: helm-charts/yunikorn/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,74 @@
+apiVersion: apps/v1beta2
```
Comment (yangwwei) [inline helm-charts/yunikorn/templates/deployment.yaml:1]: add apache license header

File: helm-charts/yunikorn/templates/rbac.yaml [hunk 7]
```diff
@@ -0,0 +1,24 @@
+apiVersion: v1
```
Comment (yangwwei) [inline helm-charts/yunikorn/templates/rbac.yaml:1]: apache license

File: helm-charts/yunikorn/values.yaml [hunk 8]
```diff
@@ -0,0 +1,98 @@
+# This is a YAML-formatted file.
```
Comment (yangwwei) [inline helm-charts/yunikorn/values.yaml:1]: apache license

File: helm-charts/yunikorn/templates/service.yaml [hunk 9]
```diff
@@ -0,0 +1,23 @@
+apiVersion: v1
```
Comment (yangwwei) [inline helm-charts/yunikorn/templates/service.yaml:1]: apache license

**Comment by kingamarton** [comment]
> Shall we remove the helm-charts from k8shim repo?
> how can we get them aligned? We don't want to have 2 dup places to maintain chart files.



> Shall we remove the helm-charts from k8shim repo?
> how can we get them aligned? We don't want to have 2 dup places to maintain chart files.

I definitely agree that we should maintain it only in one place. That was one of the reasons why I wanted to do it first in the shim repo. What do you think it wouldn't be uncomfortable for us during development to use a a repository for making the changes and building the images and a different one to install it?

File: helm-charts/index.yaml (hunk 2 above)
Comment (kingamarton) [reply helm-charts/index.yaml:6]: it takes the appVersion and api version from the Charts.yaml file.
The api version is v1 if we want to support older versions of Helm than v3.
The appVersion should be the version of the application, so I will change it to 0.8.0

**Comment by wilfred-s** [comment]
I am a huge +1 on removing the helm charts from the k8shim. I have created YUNIKORN-159 to remove the files and update the documentation when this is done.

I don't think building the images needs to move. This is just about pushing the images and the helm charts on release. The Makefile will still be there to generate the images. You can have a copy of the helm charts locally if you want to use that 

What I am currently missing is the integration into the release process. I assume that we will only push new helm charts when we release a new version. That means the tool for building a release needs to be adjusted to generate the new chart for a release.

**Comment by kingamarton** [comment]
@wilfred-s  I will open a Jira to include the chart release in the release process. After deleting the charts from the shim repo we will have to modify the tool anyways.

**Pull request merged**